linera-core/src/chain_worker/state/guard.rs

```diff
@@ -913,6 +913,18 @@ where
 
             info.requested_pending_message_bundles = messages;
         }
+        if !query.request_sent_certificate_hashes_by_heights.is_empty() {
+            let mut hashes = Vec::new();
+            for height in query.request_sent_certificate_hashes_by_heights {
+                let hash = chain
+                    .confirmed_log
```

Comment:
If `height > chain.tip_state.get().next_block_height`, we need to look it up in `preprocessed_blocks` rather than `confirmed_log`. Alternatively: `chain.block_hashes(height..=height)`.

Reply:
I see how `chain.block_hashes(height..=height)` would also work but it seems like an unnecessary complexity? I'd rather first try `confirmed_log` and - if `None` - try `preprocessed_blocks`.

Reply:
Done https://github.com/linera-io/linera-protocol/pull/4373/commits/e0d7044142d5788a4aae718a9945f78708b4ba9e